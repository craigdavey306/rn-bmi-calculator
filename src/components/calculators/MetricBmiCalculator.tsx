import { Text, View } from 'react-native';

import MetricHeightInput from '../ui/MetricHeightInput';
import WeightInput from '../ui/WeightInput';

import { MetricCalculator } from '../ui/BmiCalculator';
import MetricWeightInput from '../ui/MetricWeightInput';

type MetricBmiCalculatorProps = {
  handleHeightInput: (heightCentimeters: string) => void;
  handleWeightInput: (weightKilograms: string) => void;
} & Pick<MetricCalculator, 'heightCentimeters' | 'weightKilograms'>;

const MetricBmiCalculator: React.FC<MetricBmiCalculatorProps> = ({
  handleHeightInput,
  handleWeightInput,
  heightCentimeters,
  weightKilograms,
}): JSX.Element => {
  return (
    <View>
      <MetricHeightInput
        handleHeight={handleHeightInput}
        heightInCentimeters={heightCentimeters}
      />
      <MetricWeightInput
        handleWeightInput={handleWeightInput}
        weightInKilograms={weightKilograms}
      />
    </View>
  );
};

export default MetricBmiCalculator;
