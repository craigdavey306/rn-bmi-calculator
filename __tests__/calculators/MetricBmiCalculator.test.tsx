import 'react-native';
import renderer from 'react-test-renderer';
import { expect, it } from '@jest/globals';
import MetricBmiCalculator from '../../src/components/calculators/MetricBmiCalculator';

it('Renders the English BMI Calculator correctly', () => {
  const stub = (input: string) => {};

  const calculator = (
    <MetricBmiCalculator
      heightCentimeters=""
      weightKilograms=""
      handleHeightInput={stub}
      handleWeightInput={stub}
    />
  );

  const tree = renderer.create(calculator).toJSON();

  expect(tree).toMatchSnapshot();
});
