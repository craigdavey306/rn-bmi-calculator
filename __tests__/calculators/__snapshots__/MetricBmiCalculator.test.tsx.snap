// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the English BMI Calculator correctly 1`] = `
<View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#234e27",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 5,
          "textAlign": "center",
        }
      }
    >
      Height
    </Text>
    <View
      style={
        {
          "marginVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        nativeID="centimeters"
        style={
          {
            "color": "#000000",
            "fontSize": 13,
            "marginBottom": 4,
            "marginLeft": 3,
          }
        }
      >
        Centimeters
      </Text>
      <TextInput
        accessibilityLabelledBy="centimeters"
        accessibilityRole="none"
        accessible={true}
        autoCapitalize="none"
        keyboardType="number-pad"
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#dfeedf",
              "borderColor": "#000000",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 13,
              "paddingHorizontal": 6,
              "paddingVertical": 8,
            },
          ]
        }
        testID="centimeters:testid"
        value=""
      />
    </View>
  </View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#234e27",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 5,
          "textAlign": "center",
        }
      }
    >
      Weight
    </Text>
    <View
      style={
        {
          "marginVertical": 8,
        }
      }
    >
      <Text
        accessibilityRole="text"
        nativeID="kilograms"
        style={
          {
            "color": "#000000",
            "fontSize": 13,
            "marginBottom": 4,
            "marginLeft": 3,
          }
        }
      >
        Kilograms
      </Text>
      <TextInput
        accessibilityLabelledBy="kilograms"
        accessibilityRole="none"
        accessible={true}
        autoCapitalize="none"
        keyboardType="numeric"
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#dfeedf",
              "borderColor": "#000000",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 13,
              "paddingHorizontal": 6,
              "paddingVertical": 8,
            },
          ]
        }
        testID="kilograms:testid"
        value=""
      />
    </View>
  </View>
</View>
`;
